### PR TITLE
[High][PS] Fix PS-Overlay Widget performance issues

### PR DESCRIPTION
**Pull request by moinm3uw**
https://trello.com/c/D5LE2XI3/836-highps-fix-ps-overlay-widget-performance-issues
tick doesn't perform any heavy logic

### CODE REVIEW

File: Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp [hunk 1]
```diff
@@ -108,5 +117,5 @@ void UPSOverlayWidget::TickPlayFadeOverlayAnimation()
 void UPSOverlayWidget::SetOverlayItemsVisibility(ESlateVisibility VisibilitySlate)
 {
 	// Level is unlocked hide the blocking overlay
-	PSCOverlay->SetVisibility(VisibilitySlate);
+	this->SetVisibility(VisibilitySlate);
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp]: please remove `this->`

File: Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp [hunk 2]
```diff
@@ -61,6 +62,14 @@ void UPSOverlayWidget::SetOverlayVisibility(ESlateVisibility VisibilitySlate, bo
 void UPSOverlayWidget::NativeTick(const FGeometry& MyGeometry, float InDeltaTime)
 {
 	Super::NativeTick(MyGeometry, InDeltaTime);
+	if (PSCOverlay)
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp]: Please simplify the logic in Native Tick to the few lines:
- don't use PCSOverlay at all
- don't change visibility at all (its responsibility of other functions)

in tick, only do early return checking self visibility:
if (GetVisibility() != Visible)
{
return;
}

File: Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp (hunk 1 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp]: done with https://github.com/h4rdmol/ProgressionSystem/pull/34/commits/c64bce5118da025fdcfdf728ddc6c97e16cc2b76

File: Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp (hunk 2 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp]: done with https://github.com/h4rdmol/ProgressionSystem/pull/34/commits/7de39fb2669235d271becbcfe70dd0d18ffad97a

File: Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp [hunk 3]
```diff
@@ -108,5 +113,6 @@ void UPSOverlayWidget::TickPlayFadeOverlayAnimation()
 void UPSOverlayWidget::SetOverlayItemsVisibility(ESlateVisibility VisibilitySlate)
 {
 	// Level is unlocked hide the blocking overlay
+	SetVisibility(VisibilitySlate);
 	PSCOverlay->SetVisibility(VisibilitySlate);
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp]: Please remove changing visibility of PSCOverlay `PSCOverlay->SetVisibility(VisibilitySlate);`, as now you hide the widget itself, so no need to hide its overlay component

Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp]: done with: https://github.com/h4rdmol/ProgressionSystem/pull/34/commits/682403d36829f78f030cec4787683d9440153519

**Pull request merged**